src/legacy/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -264,6 +265,12 @@ function VisEditor(
     // This is used to sync visualization state with the url when `appState.save()` is called.
     const appState = new AppState(stateDefaults);
 
+    // Initializing appState does two things - first it translates the defaults into AppState,
+    // second it updates appState based on the url (the url trumps the defaults). This means if
+    // we update the state format at all and want to handle BWC, we must not only migrate the
+    // data stored with saved vis, but also any old state in the url.
```

Comment:
This is a great comment.

Reply:
Wish I could take credit; stolen from @Stacey-Gammon  😄